vm-setup/roles/v1aX_integration_test/tasks/remediation.yml

```diff
@@ -220,6 +220,50 @@
     delay: 20
     register: ready_hosts
     until: ready_hosts.stdout == "1"
+    
+    #### m3dt template reference check
+  - name: Get the m3dt,edit and apply a new one 
+    shell: kubectl get m3dt -n "{{ NAMESPACE }}" "{{ CLUSTER_NAME }}"-workers-template  -ojson | 
+              jq '.metadata.name="test-new-m3dt"' |  
+              jq '.|del(.metadata|.managedFields,.uid,.resourceVersion,.ownerReferences,.finalizers)' |
+              jq '.|del(.status)' | 
+              jq '.spec.templateReference="{{ CLUSTER_NAME }}-workers-template"' |
+              kubectl apply -f- 
+
+  - name: Get the m3mt,edit and apply a new one 
+    shell: kubectl get m3mt -n "{{ NAMESPACE }}" "{{ CLUSTER_NAME }}"-workers -ojson |
+              jq '.spec.template.spec.dataTemplate.name="test-new-m3dt"' | 
+              jq '.metadata.name="test-new-m3mt"' |  
+              jq '.|del(.metadata|.managedFields,.uid,.resourceVersion,.ownerReferences,.finalizers)' |
+              jq '.|del(.status)' | 
+              kubectl apply -f-
+
+  - name: Edit MachineDeployment to point to the new m3mt
```

Comment:
This is a good descriptive.
The other task names can also be improved like it as the current names are referring to the operations, not purpose.

Reply:
Thanks done